src/components/common/YourCompany/StartDate.vue

```diff
@@ -0,0 +1,232 @@
+          <div class="float-right mb-2">
+            <v-btn large color="primary" id="start-done-btn" class="mr-2" @click="onDoneClicked()">
+              <span>Done</span>
+            </v-btn>
+            <v-btn large outlined color="primary" id="start-cancel-btn" @click="onCancelClicked()">
+              <span>Cancel</span>
+            </v-btn>
+          </div>
+        </template>
+
+        <span class="mt-n2 mr-n3 float-right">
+          <!-- Correct changes button -->
+          <v-btn v-if="isCorrectionFiling && !onEditMode && !isCorrected" text color="primary"
+            id="start-changes-btn" @click="onChangeClicked()">
+            <v-icon small color="primary">mdi-pencil</v-icon>
+            <span>{{ editLabel }}</span>
+          </v-btn>
+
+          <!-- Undo changes buttons -->
+          <v-btn v-if="isCorrectionFiling && !onEditMode && isCorrected"
+          text color="primary" id="start-undo-btn" @click="onUndoClicked()">
+            <v-icon small>mdi-undo</v-icon>
+            <span>Undo</span>
+          </v-btn>
+          <v-menu v-if="isCorrectionFiling && !onEditMode && isCorrected"
+            offset-y left nudge-bottom="4" v-model="dropdown">
+            <template v-slot:activator="{ on }">
+              <v-btn text small color="primary" id="start-menu-btn" v-on="on">
+                <v-icon id="start-menu">{{dropdown ? 'mdi-menu-up' : 'mdi-menu-down'}}</v-icon>
+              </v-btn>
+            </template>
+            <v-btn text color="primary" id="more-changes-btn"
+              @click="onChangeClicked(); dropdown = false">
+              <v-icon small color="primary">mdi-pencil</v-icon>
+              <span>{{ editLabel }}</span>
+            </v-btn>
+          </v-menu>
+        </span>
+
+      </v-col>
+    </v-row>
+  </section>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop, Watch } from 'vue-property-decorator'
+import { Getter, Action } from 'vuex-class'
+import { CommonMixin, DateMixin } from '@/mixins/'
+import { DatePicker as DatePickerShared } from '@bcrs-shared-components/date-picker'
+import { ActionBindingIF } from '@/interfaces'
+
+@Component({
+  components: {
+    DatePickerShared
+  }
+})
+export default class StartDate extends Mixins(CommonMixin, DateMixin) {
+  // Global getters
+  @Getter isBenIaCorrectionFiling!: boolean
+  @Getter isFirmCorrectionFiling!: boolean
+  @Getter hasBusinessStartDateChanged!: boolean
+  @Getter getCorrectionStartDate!: string
+  @Getter getBusinessFoundingDate!: string // actually date-time
+  @Getter getCurrentDate!: string
+  @Getter getCurrentJsDate!: Date
+
+  // Global setter
+  @Action setCorrectionStartDate!: ActionBindingIF
+  @Action setValidComponent!: ActionBindingIF
+
+  /** Whether to show invalid section styling. */
+  @Prop({ default: false })
+  readonly invalidSection: boolean
+
+  protected dropdown = false
+  protected onEditMode = false
+  protected isCorrected = !!this.getCorrectionStartDate
```

Comment:
This could use the getter, too.

I'm a bit concerned that the initial value will be incorrect (if this variable is initialized before the draft correction is loaded). Does it matter, since it's set in `businessStartDate` below, which is reactive? I dunno, probably needs some testing with it initially True and False.

Reply:
I'll add this, I have just finished testing it, it does not matter since it is set in businessStartDate below.

Reply:
If it is True initially, does it get set to False below?

Reply:
It does not. I can make it do that though by setting it the same way I set it to true below.

Reply:
Sounds good.

Also, if you don't want to make any assumptions about the initial value, you can set it to null. It's kind of a 3rd value for booleans. You do have to set a value (and not undefined) or it won't be reactive and the v-ifs in the template won't see any changes to it.

It complicated and simple eh?

Reply:
Ya it really is, I like the setting it to null, the less assumptions the better probably.